src/parcsr_ls/par_ilu_setup_device.c

```diff
@@ -146,7 +179,17 @@ hypre_ILUSetupILUDevice(HYPRE_Int               ilu_type,
          hypre_CSRMatrixPermute(hypre_ParCSRMatrixDiag(A), perm_data, rqperm_data, &A_diag);
 
          /* Compute ILU0 on the device */
-         hypre_CSRMatrixILU0(A_diag);
+         if (iter_setup_type)
```

Comment:
Since the check for using rocsparse is embedded in the hypre_ILUSetupIterativeILU0Device call, the assumption is that new support for iterative ILU factorization on other devices will have the same function prototype. Is that the idea?